opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -171,11 +125,56 @@ impl HttpExporterBuilder {
             },
             None => self.exporter_config.timeout,
         };
-        let http_client = self
-            .http_config
-            .client
-            .take()
-            .ok_or(crate::Error::NoHttpClient)?;
+
+        #[allow(unused_mut)] // TODO - clippy thinks mut is not needed, but it is
```

Comment:
there is another issue about parsing env variables for timeout, I opened https://github.com/open-telemetry/opentelemetry-rust/issues/2591 to track it separate from this PR